packages/flutter/test/cupertino/context_menu_test.dart

```diff
@@ -810,4 +811,71 @@ void main() {
       expect(right.dx, lessThan(left.dx));
     });
   });
+
+  testWidgets('Is gesture conflicts of ctx_menu resolved',
+      (WidgetTester tester) async {
+    int? onPointerDownTime;
+    int? onPointerUpTime;
+    bool insideTapTriggered = false;
+    // 500ms < duration < 900ms
+    const Duration pressDuration = Duration(milliseconds: 501);
+
+    int now() => clock.now().millisecondsSinceEpoch;
+
+    await tester.pumpWidget(Listener(
+      onPointerDown: (PointerDownEvent event) => onPointerDownTime = now(),
+      onPointerUp: (PointerUpEvent event) => onPointerUpTime = now(),
+      child: CupertinoApp(
+        home: Align(
+          child: CupertinoContextMenu(
+            actions: const <CupertinoContextMenuAction>[
+              CupertinoContextMenuAction(
+                child: Text('CupertinoContextMenuAction'),
+              ),
+            ],
+            child: GestureDetector(
+              onTap: () => insideTapTriggered = true,
+              child: Container(
+                width: 200,
+                height: 200,
+                key: const Key('container'),
+                color: const Color(0xFF00FF00),
+              ),
+            ),
+          ),
+        ),
+      ),
+    ));
+
+    // Start a press on the child.
+    final TestGesture gesture = await tester.createGesture();
+    await gesture.down(tester.getCenter(find.byKey(const Key('container'))));
+    await tester.pump();
+    // simulate a press with specified duration
+    for (int i = 0; i < 100; i++) {
```

Comment:
Did you try `await tester.pump(const Duration(milliseconds: pressDuration));` first? Is this for loop more consistent?

Reply:
Yes, I tried it first. But `_openAnimation` won't reach midPoint since a `pump` only triggers one frame.

I need more frames to drive `_openAnimation` to reach the `midPoint`, so the route can be opened.

Reply:
https://github.com/flutter/flutter/blob/2c71881f505b0dc0997e959b3ac37664d221fad8/packages/flutter_test/lib/src/widget_tester.dart#L704C3-L719C4

`pumpFrames` use the same methodology like this. And it will pump another widget.

Reply:
Does it really matter that 100 frames need to be pumped, not just 1 frame over the same duration?  I wouldn't think that would affect the gesture.